src/cascadia/TerminalSettingsModel/FontConfig.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "FontConfig.h"
+#include "FontConfig.g.cpp"
+#include "TerminalSettingsSerializationHelpers.h"
+#include "JsonUtils.h"
+
+using namespace Microsoft::Terminal::Settings::Model;
+using namespace winrt::Microsoft::Terminal::Settings::Model::implementation;
+
+static constexpr std::string_view FontInfoKey{ "font" };
+static constexpr std::string_view FontFaceKey{ "face" };
+static constexpr std::string_view FontSizeKey{ "size" };
+static constexpr std::string_view FontWeightKey{ "weight" };
+static constexpr std::string_view LegacyFontFaceKey{ "fontFace" };
+static constexpr std::string_view LegacyFontSizeKey{ "fontSize" };
+static constexpr std::string_view LegacyFontWeightKey{ "fontWeight" };
+
+winrt::Microsoft::Terminal::Settings::Model::implementation::FontConfig::FontConfig(const winrt::weak_ref<Profile> sourceProfile) :
+    _sourceProfile(sourceProfile)
+{
+}
+
+winrt::com_ptr<FontConfig> FontConfig::CopyFontInfo(const winrt::com_ptr<FontConfig> source, const winrt::weak_ref<Profile> sourceProfile)
+{
+    auto fontInfo{ winrt::make_self<FontConfig>(sourceProfile) };
+    auto const sourceFontInfo = source.try_as<FontConfig>();
+    fontInfo->_FontFace = sourceFontInfo->_FontFace;
+    fontInfo->_FontSize = sourceFontInfo->_FontSize;
+    fontInfo->_FontWeight = sourceFontInfo->_FontWeight;

```

Comment:
Wait I don't understand... 
`try_as` technically, potentially returns a null pointer. It should be necessary to check the return value for nullability.
But in the end the argument is already the FontConfig type you want. You don't have to cast it anymore at this point.

I'm going to block this PR due this comment.

Reply:
You're right! Removed the casting